XamarinCommunityToolkit/Views/AvatarView.shared.cs

```diff
@@ -0,0 +1,252 @@
+using System;
+using System.ComponentModel;
+using Xamarin.Forms;
+using static System.Math;
+
+namespace XamarinCommunityToolkit.Views
+{
+    public class AvatarView : Frame
+    {
+        static readonly Color[] colors = {
+            RGB(69, 43, 103),
+            RGB(119, 78, 133),
+            RGB(211, 153, 184),
+            RGB(249, 218, 231),
+            RGB(223, 196, 208),
+            RGB(209, 158, 180),
+            RGB(171, 116, 139),
+            RGB(143, 52, 87)
+        };
+
+        static readonly Color[] textColors = {
+            RGB(53, 21, 61),
+            RGB(255, 255, 255),
+            RGB(131, 81, 102),
+            RGB(114, 50, 75)
+        };
+
+        public static readonly BindableProperty SizeProperty = BindableProperty.Create(nameof(Size), typeof(double), typeof(AvatarView), 40.0, propertyChanged: OnSizePropertyChanged);
+
+        public static new readonly BindableProperty CornerRadiusProperty = BindableProperty.Create(nameof(CornerRadius), typeof(double), typeof(AvatarView), -1.0, propertyChanged: OnSizePropertyChanged);
+
+        public static new readonly BindableProperty BackgroundColorProperty = BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(AvatarView), Color.Default, propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty SourceProperty = BindableProperty.Create(nameof(Source), typeof(ImageSource), typeof(AvatarView), propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty TextProperty = BindableProperty.Create(nameof(Text), typeof(string), typeof(AvatarView), propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty TextColorProperty = BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(AvatarView), Color.Default, propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty FontFamilyProperty = BindableProperty.Create(nameof(FontFamily), typeof(string), typeof(AvatarView), propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty FontSizeProperty = BindableProperty.Create(nameof(FontSize), typeof(double), typeof(AvatarView), -1.0, propertyChanged: OnValuePropertyChanged);
+
+        public static readonly BindableProperty FontAttributesProperty = BindableProperty.Create(nameof(FontAttributes), typeof(FontAttributes), typeof(AvatarView), FontAttributes.None, propertyChanged: OnValuePropertyChanged);
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This property is disabled.", true)]
```

Comment:
Is there another/better option than `Obsolete`?

Maybe it's me, but obsolete to me means something that worked before. Which is not the case for this. Nothing major here, just wondering

Reply:
This property is hidden from IntelliSense anyway :)
Also, **True** in obsolete will cause compile error if somebody tries to use it.